### PR TITLE
flake-info: enforce minimal nix version

### PR DESCRIPTION
**Pull request by ncfavier**
We need at least nix 2.7 (or 2.7pre_old_enough) for the nixpkgs import, so give a helpful error instead of letting nix fail.

It got a bit messy because nix versions aren't quite semver-compliant (there should be a dash before pre) so I had to strip the pre-suffix.

### CODE REVIEW

**Comment by ysndr** [comment]
Let's make this a general check, i.e. test this berfore running any nix command in https://github.com/NixOS/nixos-search/blob/26caeaf4883c13b1d9f3d47c922308709930346e/flake-info/src/bin/flake-info.rs#L206

I dont quite like that some features do work with some nix versions while other wont. It can lead to quite some bad UX imo

**Comment by ncfavier** [comment]
OK, but I need Rust assistance: if I move this code into `run_command`, how do I handle errors? That function returns a `FlakeInfoError`, should I add a new constructor `FlakeInfo(anyhow::Error)` or something?

**Comment by ysndr** [comment]
I moved it into its own module and added an error type for it
Also declare the nix version in a cargo config file

File: flake-info/src/commands/nix_check_version.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+use command_run::Command;
+use log::info;
+use semver::{Version, VersionReq};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum NixCheckError {
+    #[error("Installed nix doesn't match version requirement: {0} (required >={1})")]
```
Comment (ncfavier) [inline flake-info/src/commands/nix_check_version.rs:8]:
```suggestion
    #[error("Installed nix doesn't match version requirement: {0} (required {1})")]
```

Comment (ncfavier) [reply flake-info/src/commands/nix_check_version.rs:8]: Wait, it's my PR 😅

**Comment by ncfavier** [comment]
Looks good now, thank you

**Comment by ysndr** [comment]
have you tried it? I didnt have time to find an old nix

**Review by ysndr — APPROVED** [review]

**Comment by ncfavier** [comment]
Yes, succeeds and fails fine.

**Comment by ysndr** [comment]
image

**Pull request merged**